scripts/trim_sdk.py

```diff
@@ -70,6 +70,9 @@ def remove_unused_api_versions(resource_type):
         if resource_type in profile:
             # value is str like '2022-01-01' or SDKProfile
             value = profile[resource_type]
+            if value is None:
+                _LOGGER.info(f'{resource_type}\'s API version is None, skip')
```

Comment:
`ResourceType.MGMT_NETWORK_PRIVATEDNS: None` but we still need it.

Reply:
Why is `ResourceType.MGMT_NETWORK_PRIVATEDNS` introduced if its value is `None`?

Reply:
@necusjz to explain

Reply:
Adapt to such path conversion logic (line 665 to be exact):
https://github.com/Azure/azure-cli/blob/11c0f3ba64fc29475070a59a04536f19245f71d1/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L657-L671

`azure-mgmt-privatedns` is a kind of "unversioned sdk path".